src/interactions/slashCommands/mixins/CommandOptionWithChoices.ts

```diff
@@ -66,21 +66,23 @@ export abstract class ApplicationCommandOptionWithChoicesBase<T extends string |
 	 * Marks the option as autocompletable
 	 * @param autocomplete If this option should be autocompletable
 	 */
-	public setAutocomplete(autocomplete: boolean): Omit<this, 'addChoice' | 'addChoices'> {
-		if (typeof this.choices !== 'undefined') {
-			throw new RangeError('Autocomplete and choices are mutually exclusive to each other.');
-		}
-
+	public setAutocomplete<U extends boolean>(
+		autocomplete: U,
+	): U extends true ? Omit<this, 'addChoice' | 'addChoices'> : this {
```

Comment:
You need the `this & Pick<ApplicationCommandOptionWithChoicesBase<T>, 'addChoice' | 'addChoices'>`

Reply:
What's the point of that? `this` already includes addChoice and addChoices.

Reply:
`this` is not the same as `ApplicationCommandOptionWithChoicesBase`, this represents the dynamic type of the class, whereas `ApplicationCommandOptionWithChoicesBase` is the static type of the class. I made the same mistake earlier.